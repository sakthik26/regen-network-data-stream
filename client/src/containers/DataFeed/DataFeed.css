.project-title {
  width: 100%;
  text-align: center;
  font-size: 24px;
  margin: 20px 0;
}

.posts-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  gap: 20px;
  width: 60%;
  margin: 0 auto;
}
