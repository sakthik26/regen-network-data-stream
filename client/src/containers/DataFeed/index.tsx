import React, { useState, useEffect } from 'react';
import Posts from '../../components/Posts';
import { fetchPosts } from '../../utils/services';
import './DataFeed.css';
import { projectId as defaultId } from '../../constants';
import { useLocation, useParams } from 'react-router-dom';
import { Post } from '../../types';

// Container that lists all the data posts from the project 
const DataFeed: React.FC = () => {
  const { projectId } = useParams<{ projectId: string }>();
  const location = useLocation();
  const state = location.state as { projectName: string };
  const projectTitle = state?.projectName ?? '';
  const [posts, setPosts] = useState<Post[]>([]);

  useEffect(() => {
    // Fetch the posts based on the project
    const getPosts = async () => {
      try {
        const fetchedPosts = await fetchPosts(projectId || defaultId);
        setPosts(fetchedPosts);
      } catch (error) {
        console.error('Failed to fetch posts:', error);
      }
    };

    getPosts();
  }, []);

  return (
    <>
      <div className="project-title">{projectTitle}</div>
      <div className="posts-container">
        {posts.map((post) => (
          <Posts id={post.id} title={post.title} createdAt={post.createdAt} />
        ))}
      </div>
    </>
  );
};

export default DataFeed;
