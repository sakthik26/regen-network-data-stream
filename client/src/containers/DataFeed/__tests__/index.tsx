import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DataFeed from '../index';

jest.mock('../../../utils/services', () => ({
  fetchPosts: jest.fn(),
}));

describe('DataFeed component', () => {
  test('renders correctly with empty posts', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <DataFeed />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly with posts', () => {
    const posts = [
      { id: 1, title: 'Post 1', createdAt: '2024-02-11' },
      { id: 2, title: 'Post 2', createdAt: '2024-02-10' },
    ];
    jest.spyOn(React, 'useState').mockReturnValueOnce([posts, jest.fn()]);

    const { asFragment } = render(
      <MemoryRouter>
        <DataFeed />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
