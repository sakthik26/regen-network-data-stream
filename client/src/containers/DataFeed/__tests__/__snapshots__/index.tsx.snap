// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataFeed component renders correctly with empty posts 1`] = `
<DocumentFragment>
  <div
    class="project-title"
  />
  <div
    class="posts-container"
  />
</DocumentFragment>
`;

exports[`DataFeed component renders correctly with posts 1`] = `
<DocumentFragment>
  <div
    class="project-title"
  />
  <div
    class="posts-container"
  />
</DocumentFragment>
`;
